retrofit/src/main/java/retrofit/RxSupport.java

```diff
@@ -21,24 +24,35 @@
   RxSupport() {
   }
 
-  Observable createRequestObservable(final Invoker invoker) {
-    return Observable.create(new Observable.OnSubscribe<Object>() {
-      @Override public void call(final Subscriber<? super Object> subscriber) {
-        invoker.invoke(new Invoker.Callback() {
-          @Override public void next(Object o) {
-            if (!subscriber.isUnsubscribed()) {
-              subscriber.onNext(o);
-              subscriber.onCompleted();
-            }
-          }
+  Observable createRequestObservable(final Invoker invoker, final RequestBuilder requestBuilder,
+      final RequestInterceptor requestInterceptor) {
+    return Observable.defer(new Func0<Observable<Object>>() {
```

Comment:
In general, we try very-hard to minimize the per-request allocations and RxJava is extremely liberal in its allocations. Avoiding it as much as possible (which is ironic) is usually a good practice.

Your larger example exercises OkHttp's synchronous API which we can't use. All interactions will be async such that the threading model is pushed down into OkHttp which allows for optimizations in both controlling per-server connections as well as more effective interleaving with HTTP/2.

In general this code will churn a lot in the next 6 weeks, so I'm inclined to just defer (ha!) changing it.